lib/skate/settings/test_group.ex

```diff
@@ -12,12 +12,20 @@ defmodule Skate.Settings.TestGroup do
 
   defstruct [:id, :name, users: []]
 
-  @spec create(String.t()) :: __MODULE__.t()
+  @spec create(String.t()) :: {:ok, t()} | {:error, Ecto.Changeset.t()}
   def create(name) do
-    %DbTestGroup{name: name}
-    |> Skate.Repo.insert!()
-    |> Skate.Repo.preload(:users)
-    |> convert_from_db_test_group()
+    case %DbTestGroup{}
+         |> DbTestGroup.changeset(%{name: name})
+         |> Skate.Repo.insert() do
+      {:ok, new_test_group} ->
+        {:ok,
+         new_test_group
+         |> Skate.Repo.preload(:users)
+         |> convert_from_db_test_group()}
+
+      {:error, errored_changeset} ->
+        {:error, errored_changeset}
```

Comment:
**suggestion:** should also write tests for this error case in `Skate.Settings.TestGroupTest`.